Headers/DebugServer2/Utils/Bits.h

```diff
@@ -0,0 +1,88 @@
+//
+// Copyright (c) 2014-present, Facebook, Inc.
+// All rights reserved.
+//
+// This source code is licensed under the University of Illinois/NCSA Open
+// Source License found in the LICENSE file in the root directory of this
+// source tree. An additional grant of patent rights can be found in the
+// PATENTS file in the same directory.
+//
+
+#ifndef __DebugServer2_Utils_Bits_h
+#define __DebugServer2_Utils_Bits_h
+
+#include <type_traits>
+
+#include "DebugServer2/Utils/CompilerSupport.h"
+
+namespace ds2 {
+namespace Utils {
+namespace {
+// We apply the builtin bit functions on chunks of sizeof(int) * 8 bits.
+// However, we can't do full shifts of this size because only the least
+// signifitive bits are used and clang complains, so we shift twice half
```

Comment:
> it seems that clang is not smart enough

ah, yeah

> another option is to avoid templates…

yeah.

I've definitely caused more than enough thrash on this. I'm going get out of your way now. :bow: